webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/services/impl/GitUtilsImpl.scala

```diff
@@ -61,43 +61,48 @@ import com.normation.zio._
  * If a .git repository is found in it, it is considered as the repository
  * to use, else if none is found, one is created.
  */
-class GitRepositoryProviderImpl(techniqueDirectoryPath: String) extends GitRepositoryProvider { //we expect to have a .git here
-  /**
-   * Check for root package existence
-   */
-  private def checkPackageDirectory(dir: File): IOResult[Unit] = {
-    if (!dir.exists) {
-      Inconsistancy("Directory %s does not exist, how do you want that I read policy package in it?".format(dir)).fail
-    } else if (!dir.canRead) {
-      Inconsistancy("Directory %s is not readable, how do you want that I read policy package in it?".format(dir)).fail
-    } else UIO.unit
-  }
+class GitRepositoryProviderImpl(override val db: Repository) extends GitRepositoryProvider { //we expect to have a .git here
+  override val git = new Git(db)
+}
 
+object GitRepositoryProviderImpl {
   /**
-   * Ckeck git repos existence.
-   * If no git repos is found, create one.
+   * Use the provided full path as the root directory for given git provider.
    */
-  private def checkGitRepos(root:File) : IOResult[Repository] = {
-    IOResult.effect {
-      val db = (new FileRepositoryBuilder().setWorkTree(root).build).asInstanceOf[FileRepository]
-      if(!db.getConfig.getFile.exists) {
-        GitRepositoryLogger.logEffect.info(s"Git directory was not initialised: create a new git repository into folder '${root.getAbsolutePath}' and add all its content as initial release")
-        db.create()
-        val git = new Git(db)
-        git.add.addFilepattern(".").call
-        git.commit.setMessage("initial commit").call
+  def make(gitRootPath: String): IOResult[GitRepositoryProviderImpl] = {
+    /**
```

Comment:
This is the other interesting change, where the creation of git repo is effectfull, but once we have it, we just give it to `new GitRepositoryProviderImpl()`